futurex_openedx_extensions/dashboard/details/learners.py

```diff
@@ -156,14 +156,27 @@ def get_learners_queryset(
     """
     tenant_sites = get_tenants_sites(fx_permission_info['permitted_tenant_ids'])
 
+    orgs = list(set(
+        fx_permission_info['view_allowed_full_access_orgs'] + fx_permission_info['view_allowed_course_access_orgs']
+    ))
+
+    if include_staff:
+        is_staff_queryset = Q(Value(False, output_field=BooleanField()))
```

Comment:
Are we sure this pattern is good?

We can use the following pattern instead:

```py
queryset = foo_bar()

if not include_staff:
    queryset = queryset.exclude(check_staff_exist_queryset(...))

return queryset.values('user_id')

Reply:
I belive it's the better pattern. Because we need the optional exclusion within an `Exists` operation. If not, then this code:
```python
queryset = queryset.filter(
        Exists(
            UserSignupSource.objects.filter(
                user_id=OuterRef('id'),
                site__in=tenant_sites,
            ).exclude(
                Q(user__is_superuser=True) | Q(user__is_staff=True) | Q(user__is_active=False)
            ).exclude(
                is_staff_queryset
            ).values('user_id')
        )
    )
```

will be:
```python
if include_staff:
    queryset = queryset.filter(
            Exists(
                UserSignupSource.objects.filter(
                    user_id=OuterRef('id'),
                    site__in=tenant_sites,
                ).exclude(
                    Q(user__is_superuser=True) | Q(user__is_staff=True) | Q(user__is_active=False)
                ).exclude(
                    check_staff_exist_queryset(ref_user_id='user_id', ref_org=orgs, ref_course_id=None)
                ).values('user_id')
            )
        )
    )
else:
    queryset = queryset.filter(
            Exists(
                UserSignupSource.objects.filter(
                    user_id=OuterRef('id'),
                    site__in=tenant_sites,
                ).exclude(
                    Q(user__is_superuser=True) | Q(user__is_staff=True) | Q(user__is_active=False)
                ).values('user_id')
            )
        )
    )

```

Reply:
Understood! Thanks @shadinaif!